lib/Sema/TypeCheckRequestFunctions.cpp

```diff
@@ -40,14 +40,10 @@ InheritedTypeRequest::evaluate(
     options |= TypeResolutionFlags::AllowUnavailableProtocol;
   }
 
-  DependentGenericTypeResolver protoResolver;
-  GenericTypeToArchetypeResolver archetypeResolver(dc);
-  GenericTypeResolver *resolver;
-  if (isa<ProtocolDecl>(dc)) {
-    resolver = &protoResolver;
-  } else {
-    resolver = &archetypeResolver;
-  }
+  // FIXME: This should be part of the request!
+  TypeResolution resolution =
+    isa<ProtocolDecl>(dc) ? TypeResolution::forStructural(dc)
+                          : TypeResolution::forContextual(dc);
```

Comment:
Should be interface instead

Reply:
The plan here is to make the `TypeResolutionStage` part of the request input. One will be able to ask for the "inherited type at the structural stage", for example, and that's a different request from "inherited type at the interface stage".

Reply:
FWIW, this is handled by https://github.com/apple/swift/pull/18913